testing.md

```diff
@@ -61,5 +61,71 @@ The following query is sufficient to test Native Query as it exercises mutliple
 | laptop | 3257.093 |
 | envelopes | 77.661 |
 
+# Manual Testing (for Direct Query)
 
+At this time, Direct Query must be tested semi-manually by reviewers, until such time as the Power Query
+SDK uses the same Mashup Engine as Power BI.
 
+Each Direct Query we want to run is stored in the `resources/direct_query` directory
+
+1. Ensure you have a local ADF running.
+- Navigate to the `mongo-powerbi-connector` repository locally.
+- Run the ADF script via `./resources/run_adf start`
+- Load data
+  - If you have `mongoimport.exe` already installed, you can skip the downloading it. Otherwise
+    ```
+    MONGO_TOOLS_VERSION=100.6.1 curl -L -o mongodb-tools.zip \
+    https://fastdl.mongodb.org/tools/db/mongodb-database-tools-windows-x86_64-$MONGO_TOOLS_VERSION.zip
+    unzip -jn mongodb-tools.zip
+    chmod +x mongoimport.exe
+    ```
+  - Import the data
+    ```
+    ./mongoimport.exe --uri="mongodb://localhost:28017/reports" --drop resources/integration_test/testdata/transforms.json
+    ./mongoimport.exe --uri="mongodb://localhost:28017/reports" --drop resources/integration_test/testdata/table_ops.json
+    ```
+  - Set the ADF schema
+    ```
+    MONGOSH=$(find ./local_adf/ | grep mongo.exe | head -1)
+    # Replace with the namespace you are adding
+    $MONGOSH  -u mhuser -p pencil --eval 'db.runCommand({sqlGenerateSchema: 1, sampleNamespaces: ["reports.transforms", "reports.table_ops"], setSchemas: true})' localhost/admin
+    ```
+
+2. Build the connector. The easiest way to do this is in VSCode, using the Power Query SDk.
+- Navigate to the `connector` directory.
+- `code .` to open the project in VSCode.
+- Open `MongoDBAtlasODBC.pq`
+- On the editor window, right click anywhere in the editor window for the `MongoDBAtlasODBC.pq` file and choose *Evaluate current power query file*
+- Copy `bin\AnyCPU\Debug\connector.mez` to the Microsoft Power BI custom connector folder, located in `C:\Users\<username>\Documents\Power BI Desktop\Custom Connectors`.
+
+3. Start Power BI.
+4. Choose *Get Data*
+5. Search for "mongodb" in the search box, and select the *MongoDB Atlas SQL (Beta)* connector.
+6. Connect to your local ADF.
+- The URI should be `mongodb://localhost`
+- Make sure to select the Direct Query radio button instead of Import
+- The database is `reports`.
+7. When the data explorer comes up, select the `reports` database and checkbox both collections:
+- `transforms`
+- `table_ops`
+8. Wait a very long time for both collections to load
```

Comment:
😆 While true, we may want to avoid editorializing to avoid poisoning other people's perceptions.